[Bug]: `samplers/test_logprobs.py` fail on H100

### Your current environment

```text
Collecting environment information...
PyTorch version: 2.3.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.29.5
Libc version: glibc-2.35

Python version: 3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:36:13) [GCC 12.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-107-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA H100 80GB HBM3
GPU 1: NVIDIA H100 80GB HBM3
GPU 2: NVIDIA H100 80GB HBM3
GPU 3: NVIDIA H100 80GB HBM3
GPU 4: NVIDIA H100 80GB HBM3
GPU 5: NVIDIA H100 80GB HBM3
GPU 6: NVIDIA H100 80GB HBM3
GPU 7: NVIDIA H100 80GB HBM3

Nvidia driver version: 550.54.15
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.7
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      52 bits physical, 57 bits virtual
Byte Order:                         Little Endian
CPU(s):                             192
On-line CPU(s) list:                0-191
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Xeon(R) Platinum 8474C
CPU family:                         6
Model:                              143
Thread(s) per core:                 2
Core(s) per socket:                 48
Socket(s):                          2
Stepping:                           8
CPU max MHz:                        3800.0000
CPU min MHz:                        800.0000
BogoMIPS:                           4200.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          4.5 MiB (96 instances)
L1i cache:                          3 MiB (96 instances)
L2 cache:                           192 MiB (96 instances)
L3 cache:                           195 MiB (2 instances)
NUMA node(s):                       2
NUMA node0 CPU(s):                  0-47,96-143
NUMA node1 CPU(s):                  48-95,144-191
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] mypy==1.9.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] nvidia-nccl-cu12==2.20.5
[pip3] sentence-transformers==3.0.1
[pip3] torch==2.3.0
[pip3] transformers==4.41.2
[pip3] triton==2.3.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-nccl-cu12          2.20.5                   pypi_0    pypi
[conda] sentence-transformers     3.0.1                    pypi_0    pypi
[conda] torch                     2.3.0                    pypi_0    pypi
[conda] transformers              4.41.2                   pypi_0    pypi
[conda] triton                    2.3.0                    pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.5.1
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0	GPU1	GPU2	GPU3	GPU4	GPU5	GPU6	GPU7	NIC0	NIC1	CPU Affinity	NUMA Affinity	GPU NUMA ID
GPU0	 X 	NV18	NV18	NV18	NV18	NV18	NV18	NV18	PIX	PIX	0-47,96-143	0		N/A
GPU1	NV18	 X 	NV18	NV18	NV18	NV18	NV18	NV18	SYS	SYS	0-47,96-143	0		N/A
GPU2	NV18	NV18	 X 	NV18	NV18	NV18	NV18	NV18	SYS	SYS	0-47,96-143	0		N/A
GPU3	NV18	NV18	NV18	 X 	NV18	NV18	NV18	NV18	SYS	SYS	0-47,96-143	0		N/A
GPU4	NV18	NV18	NV18	NV18	 X 	NV18	NV18	NV18	SYS	SYS	48-95,144-191	1		N/A
GPU5	NV18	NV18	NV18	NV18	NV18	 X 	NV18	NV18	SYS	SYS	48-95,144-191	1		N/A
GPU6	NV18	NV18	NV18	NV18	NV18	NV18	 X 	NV18	SYS	SYS	48-95,144-191	1		N/A
GPU7	NV18	NV18	NV18	NV18	NV18	NV18	NV18	 X 	SYS	SYS	48-95,144-191	1		N/A
NIC0	PIX	SYS	SYS	SYS	SYS	SYS	SYS	SYS	 X 	PIX				
NIC1	PIX	SYS	SYS	SYS	SYS	SYS	SYS	SYS	PIX	 X 				

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
```


### 🐛 Describe the bug

The `tests/samplers/test_logprobs.py` are failing on `main` branch when run on an H100 GPU:

```
$ python -m pytest tests/samplers/test_logprobs.py::test_get_prompt_logprobs
=================================================================================================== test session starts ===================================================================================================
platform linux -- Python 3.11.9, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/zrltpa/vllm
configfile: pyproject.toml
plugins: shard-0.1.2, forked-1.6.0, asyncio-0.23.7, rerunfailures-14.0, anyio-4.4.0
asyncio: mode=Mode.STRICT
collected 8 items                                                                                                                                                                                                         
Running 8 items in this shard

tests/samplers/test_logprobs.py .F...F..                                                                                                                                                                            [100%]

======================================================================================================== FAILURES =========================================================================================================
________________________________________________________________________________ test_get_prompt_logprobs[True-6-4-half-facebook/opt-125m] ________________________________________________________________________________

hf_runner = <class 'tests.conftest.HfRunner'>, vllm_runner = <class 'tests.conftest.VllmRunner'>, model = 'facebook/opt-125m', dtype = 'half', chunked_prefill_token_size = 4, num_top_logprobs = 6, detokenize = True
example_prompts = ['vLLM is a high-throughput and memory-efficient inference and serving engine for LLMs.\n', 'Briefly describe the majo...me.\n', 'Analyze the impact of the COVID-19 pandemic on global economic structures and future business models.\n', ...]

    @pytest.mark.parametrize("model", MODELS)
    @pytest.mark.parametrize("dtype", ["half"])
    @pytest.mark.parametrize("chunked_prefill_token_size", [1, 4, 16, -1])
    @pytest.mark.parametrize("num_top_logprobs", [6])  # 32000 == vocab_size
    @pytest.mark.parametrize("detokenize", [True, False])
    def test_get_prompt_logprobs(
        hf_runner,
        vllm_runner,
        model,
        dtype,
        chunked_prefill_token_size: int,
        num_top_logprobs: int,
        detokenize: bool,
        example_prompts,
    ):
        max_num_seqs = 256
        enable_chunked_prefill = False
        max_num_batched_tokens = None
        if chunked_prefill_token_size != -1:
            enable_chunked_prefill = True
            max_num_seqs = min(chunked_prefill_token_size, max_num_seqs)
            max_num_batched_tokens = chunked_prefill_token_size
    
        max_tokens = 5
        with hf_runner(model, dtype=dtype) as hf_model:
            hf_logprobs = hf_model.generate_greedy_logprobs(
                example_prompts,
                max_tokens=max_tokens,
            )
    
        with vllm_runner(
                model,
                dtype=dtype,
                max_logprobs=num_top_logprobs,
                enable_chunked_prefill=enable_chunked_prefill,
                max_num_batched_tokens=max_num_batched_tokens,
                max_num_seqs=max_num_seqs,
        ) as vllm_model:
            vllm_sampling_params = SamplingParams(max_tokens=max_tokens,
                                                  logprobs=num_top_logprobs,
                                                  prompt_logprobs=num_top_logprobs,
                                                  temperature=0.0,
                                                  detokenize=detokenize)
            vllm_results = vllm_model.model.generate(
                example_prompts, sampling_params=vllm_sampling_params)
    
        # Test whether logprobs are included in the results.
        for result in vllm_results:
            assert result.prompt_logprobs is not None
            assert result.outputs[0].logprobs is not None
            assert len(result.outputs[0].logprobs) == max_tokens
            for logprobs in result.outputs[0].logprobs:
                assert len(logprobs) == num_top_logprobs
            output_text = result.outputs[0].text
            output_string_from_most_likely_tokens_lst: List[str] = []
            for top_logprobs in result.outputs[0].logprobs:
                top_logprob = next(iter(top_logprobs.values()))
                output_string_from_most_likely_tokens_lst.append(
                    top_logprob.decoded_token)
    
            if detokenize:
                output_string_from_most_likely_tokens = "".join(
                    output_string_from_most_likely_tokens_lst)
                assert output_text == output_string_from_most_likely_tokens, (
                    "The output text from the top logprob for each token position "
                    "should be the same as the output text in the result.")
            else:
                assert output_text == ''
                assert output_string_from_most_likely_tokens_lst == ([None] *
                                                                     max_tokens)
    
            # The first prompt logprob is always None
            assert result.prompt_logprobs[0] is None
            for prompt_logprobs in result.prompt_logprobs[1:]:
                # If the prompt token is not included in the top X
                # logprob, it can return 1 more data
                assert (len(prompt_logprobs) == num_top_logprobs
                        or len(prompt_logprobs) == num_top_logprobs + 1)
    
        # Test whether prompt logprobs are consistent with HF
        for vllm_result, hf_logprob in zip(vllm_results, hf_logprobs):
            # Check prompt logprobs
            # The first prompt logprob is always None, so we compare it from 1:.
            vllm_prompt_logprobs = vllm_result.prompt_logprobs[1:]
            for i, vllm_prompt_logprob_dict in enumerate(vllm_prompt_logprobs):
                for token_id, logprob in vllm_prompt_logprob_dict.items():
>                   torch.testing.assert_close(logprob.logprob,
                                               hf_logprob[0][i][token_id].item(),
                                               atol=1e-2,
                                               rtol=1e-2)
E                   AssertionError: Scalars are not close!
E                   
E                   Expected -1.3945362567901611 but got -1.432058572769165.
E                   Absolute difference: 0.037522315979003906 (up to 0.01 allowed)
E                   Relative difference: 0.026906662194190602 (up to 0.01 allowed)

tests/samplers/test_logprobs.py:99: AssertionError
-------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------
INFO 07-13 00:09:30 config.py:778] Chunked prefill is enabled (EXPERIMENTAL).
INFO 07-13 00:09:30 llm_engine.py:174] Initializing an LLM engine (v0.5.1) with config: model='facebook/opt-125m', speculative_config=None, tokenizer='facebook/opt-125m', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, rope_scaling=None, rope_theta=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.float16, max_seq_len=1024, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, quantization_param_path=None, device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None), seed=0, served_model_name=facebook/opt-125m, use_v2_block_manager=False, enable_prefix_caching=False)
INFO 07-13 00:09:31 weight_utils.py:218] Using model weights format ['*.bin']
INFO 07-13 00:09:31 model_runner.py:266] Loading model weights took 0.2390 GB
INFO 07-13 00:09:31 gpu_executor.py:86] # GPU blocks: 129168, # CPU blocks: 7281
INFO 07-13 00:09:31 model_runner.py:1007] Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
INFO 07-13 00:09:31 model_runner.py:1011] CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
INFO 07-13 00:09:31 model_runner.py:1208] Graph capturing finished in 0 secs.
-------------------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------------------
Processed prompts: 100%|██████████| 8/8 [00:00<00:00, 24.47it/s, est. speed input: 514.15 toks/s, output: 122.41 toks/s]
_______________________________________________________________________________ test_get_prompt_logprobs[False-6-4-half-facebook/opt-125m] ________________________________________________________________________________

hf_runner = <class 'tests.conftest.HfRunner'>, vllm_runner = <class 'tests.conftest.VllmRunner'>, model = 'facebook/opt-125m', dtype = 'half', chunked_prefill_token_size = 4, num_top_logprobs = 6, detokenize = False
example_prompts = ['vLLM is a high-throughput and memory-efficient inference and serving engine for LLMs.\n', 'Briefly describe the majo...me.\n', 'Analyze the impact of the COVID-19 pandemic on global economic structures and future business models.\n', ...]

    @pytest.mark.parametrize("model", MODELS)
    @pytest.mark.parametrize("dtype", ["half"])
    @pytest.mark.parametrize("chunked_prefill_token_size", [1, 4, 16, -1])
    @pytest.mark.parametrize("num_top_logprobs", [6])  # 32000 == vocab_size
    @pytest.mark.parametrize("detokenize", [True, False])
    def test_get_prompt_logprobs(
        hf_runner,
        vllm_runner,
        model,
        dtype,
        chunked_prefill_token_size: int,
        num_top_logprobs: int,
        detokenize: bool,
        example_prompts,
    ):
        max_num_seqs = 256
        enable_chunked_prefill = False
        max_num_batched_tokens = None
        if chunked_prefill_token_size != -1:
            enable_chunked_prefill = True
            max_num_seqs = min(chunked_prefill_token_size, max_num_seqs)
            max_num_batched_tokens = chunked_prefill_token_size
    
        max_tokens = 5
        with hf_runner(model, dtype=dtype) as hf_model:
            hf_logprobs = hf_model.generate_greedy_logprobs(
                example_prompts,
                max_tokens=max_tokens,
            )
    
        with vllm_runner(
                model,
                dtype=dtype,
                max_logprobs=num_top_logprobs,
                enable_chunked_prefill=enable_chunked_prefill,
                max_num_batched_tokens=max_num_batched_tokens,
                max_num_seqs=max_num_seqs,
        ) as vllm_model:
            vllm_sampling_params = SamplingParams(max_tokens=max_tokens,
                                                  logprobs=num_top_logprobs,
                                                  prompt_logprobs=num_top_logprobs,
                                                  temperature=0.0,
                                                  detokenize=detokenize)
            vllm_results = vllm_model.model.generate(
                example_prompts, sampling_params=vllm_sampling_params)
    
        # Test whether logprobs are included in the results.
        for result in vllm_results:
            assert result.prompt_logprobs is not None
            assert result.outputs[0].logprobs is not None
            assert len(result.outputs[0].logprobs) == max_tokens
            for logprobs in result.outputs[0].logprobs:
                assert len(logprobs) == num_top_logprobs
            output_text = result.outputs[0].text
            output_string_from_most_likely_tokens_lst: List[str] = []
            for top_logprobs in result.outputs[0].logprobs:
                top_logprob = next(iter(top_logprobs.values()))
                output_string_from_most_likely_tokens_lst.append(
                    top_logprob.decoded_token)
    
            if detokenize:
                output_string_from_most_likely_tokens = "".join(
                    output_string_from_most_likely_tokens_lst)
                assert output_text == output_string_from_most_likely_tokens, (
                    "The output text from the top logprob for each token position "
                    "should be the same as the output text in the result.")
            else:
                assert output_text == ''
                assert output_string_from_most_likely_tokens_lst == ([None] *
                                                                     max_tokens)
    
            # The first prompt logprob is always None
            assert result.prompt_logprobs[0] is None
            for prompt_logprobs in result.prompt_logprobs[1:]:
                # If the prompt token is not included in the top X
                # logprob, it can return 1 more data
                assert (len(prompt_logprobs) == num_top_logprobs
                        or len(prompt_logprobs) == num_top_logprobs + 1)
    
        # Test whether prompt logprobs are consistent with HF
        for vllm_result, hf_logprob in zip(vllm_results, hf_logprobs):
            # Check prompt logprobs
            # The first prompt logprob is always None, so we compare it from 1:.
            vllm_prompt_logprobs = vllm_result.prompt_logprobs[1:]
            for i, vllm_prompt_logprob_dict in enumerate(vllm_prompt_logprobs):
                for token_id, logprob in vllm_prompt_logprob_dict.items():
>                   torch.testing.assert_close(logprob.logprob,
                                               hf_logprob[0][i][token_id].item(),
                                               atol=1e-2,
                                               rtol=1e-2)
E                   AssertionError: Scalars are not close!
E                   
E                   Expected -1.3945362567901611 but got -1.432058572769165.
E                   Absolute difference: 0.037522315979003906 (up to 0.01 allowed)
E                   Relative difference: 0.026906662194190602 (up to 0.01 allowed)

tests/samplers/test_logprobs.py:99: AssertionError
-------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------
INFO 07-13 00:09:49 config.py:778] Chunked prefill is enabled (EXPERIMENTAL).
INFO 07-13 00:09:49 llm_engine.py:174] Initializing an LLM engine (v0.5.1) with config: model='facebook/opt-125m', speculative_config=None, tokenizer='facebook/opt-125m', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, rope_scaling=None, rope_theta=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.float16, max_seq_len=1024, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, quantization_param_path=None, device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None), seed=0, served_model_name=facebook/opt-125m, use_v2_block_manager=False, enable_prefix_caching=False)
INFO 07-13 00:09:50 weight_utils.py:218] Using model weights format ['*.bin']
INFO 07-13 00:09:50 model_runner.py:266] Loading model weights took 0.2390 GB
INFO 07-13 00:09:50 gpu_executor.py:86] # GPU blocks: 129165, # CPU blocks: 7281
INFO 07-13 00:09:50 model_runner.py:1007] Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
INFO 07-13 00:09:50 model_runner.py:1011] CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
INFO 07-13 00:09:51 model_runner.py:1208] Graph capturing finished in 0 secs.
-------------------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------------------
Processed prompts: 100%|██████████| 8/8 [00:00<00:00, 41.18it/s, est. speed input: 865.03 toks/s, output: 205.95 toks/s]
==================================================================================================== warnings summary =====================================================================================================
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[True-6-1-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[True-6-4-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[True-6-16-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[True-6--1-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[False-6-1-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[False-6-4-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[False-6-16-half-facebook/opt-125m]
tests/samplers/test_logprobs.py::test_get_prompt_logprobs[False-6--1-half-facebook/opt-125m]
  /home/zrltpa/miniforge3/envs/dev-env/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================================= short test summary info =================================================================================================
FAILED tests/samplers/test_logprobs.py::test_get_prompt_logprobs[True-6-4-half-facebook/opt-125m] - AssertionError: Scalars are not close!
FAILED tests/samplers/test_logprobs.py::test_get_prompt_logprobs[False-6-4-half-facebook/opt-125m] - AssertionError: Scalars are not close!
======================================================================================== 2 failed, 6 passed, 8 warnings in 45.24s =========================================================================================
```
Looks like precision issue imo.

--- a/tests/samplers/test_logprobs.py
+++ b/tests/samplers/test_logprobs.py
@@ -11,7 +11,8 @@ MODELS = ["facebook/opt-125m"]
 
 
 @pytest.mark.parametrize("model", MODELS)
-@pytest.mark.parametrize("dtype", ["half"])
+@pytest.mark.parametrize("dtype",
+                         ["float"])  # needed for comparing logprobs with HF
 @pytest.mark.parametrize("chunked_prefill_token_size", [1, 4, 16, -1])
 @pytest.mark.parametrize("num_top_logprobs", [6])  # 32000 == vocab_size
 @pytest.mark.parametrize("detokenize", [True, False])

--- a/vllm/attention/ops/prefix_prefill.py
+++ b/vllm/attention/ops/prefix_prefill.py
@@ -687,6 +687,12 @@ if triton.__version__ >= "2.1.0":
 
         cap = current_platform.get_device_capability()
         BLOCK = 128 if cap[0] >= 8 else 64
+
+        # need to reduce num. blocks when using fp32
+        # due to increased use of GPU shared memory
+        if q.dtype == torch.float32:
+            BLOCK = BLOCK // 2
+
         # shape constraints
         Lq, Lk, Lv = q.shape[-1], k.shape[-1], v.shape[-1]
         assert Lq == Lk and Lk == Lv

--- a/vllm/attention/ops/prefix_prefill.py
+++ b/vllm/attention/ops/prefix_prefill.py
@@ -690,7 +690,7 @@ if triton.__version__ >= "2.1.0":
 
         # need to reduce num. blocks when using fp32
         # due to increased use of GPU shared memory
-        if q.dtype == torch.float32:
+        if q.dtype is torch.float32:
             BLOCK = BLOCK // 2
 
         # shape constraints

